### PR TITLE
[SPARK-43542][SS] Define a new error class and apply for the case where streaming query fails due to concurrent run of streaming query with same checkpoint

### PR DESCRIPTION
**Pull request by ericm-db**
### What changes were proposed in this pull request?

We are migrating to a new error framework in order to surface errors in a friendlier way to customers. This PR defines a new error class specifically for when there are concurrent updates to the log for the same batch ID

### Why are the changes needed?

This gives more information to customers, and allows them to filter in a better way

### How was this patch tested?

There is an existing test to check the error message upon this condition. Because we are only changing the error type, and not the error message, this test is sufficient.

### CODE REVIEW

File: core/src/main/resources/error/error-classes.json [hunk 1]
```diff
@@ -202,6 +202,13 @@
       "Another instance of this query was just started by a concurrent session."
     ]
   },
+  "CONCURRENT_STREAM_LOG_UPDATE" : {
+    "message" : [
+      "Concurrent update to the log. Multiple streaming jobs detected for <batchId>. ",
```
Comment (MaxGekk) [inline core/src/main/resources/error/error-classes.json:207]: The space at the end is redundant. Spark concatenates multi-lines by `\n`.

**Review by MaxGekk — APPROVED** [review]
LGTM except of a minor comment.

**Comment by MaxGekk** [comment]
@ericm-db Could you allow GitHub actions in your fork and re-trigger GAs, please.

**Comment by ericm-db** [comment]
Thanks for the review! I've made the changes, and I think it's ready to merge now @MaxGekk @HeartSaVioR

**Comment by MaxGekk** [comment]
+1, LGTM. Merging to master.
Thank you, @ericm-db.

**Comment by MaxGekk** [comment]
@ericm-db Congratulations with your first contribution to Apache Spark!

**Pull request closed without merging**